naptime/src/main/scala/org/coursera/naptime/resources/resources.scala

```diff
@@ -165,3 +169,58 @@ trait CollectionResource[ParentResource <: Resource[_], K, M] extends Resource[M
 trait TopLevelCollectionResource[K, M] extends CollectionResource[RootResource, K, M] {
   override val parentResource: RootResource = RootResource
 }
+
+/**
+ * If the resource is a courier model, extending from [[CourierCollectionResource]] removes some boilerplate.
+ *
+ * Aside from automating some of the boilerplate away, it functions the same as a standard
+ * [[CollectionResource]]. If the resource is a top level resource, check out the
+ * [[TopLevelCourierCollectionResource]] abstract class.
+ *
+ * To configure customized fields, use the following snippet:
+ *
+ * {{{
+ *   override implicit lazy val Fields = super.FIelds.withRelated(...)
+ * }}}
+ *
+ * @param kf The [[KeyFormat]] for the key type in the resource.
+ * @param classTag The concrete class tag for the value type in the resource.
+ * @tparam ParentResource The parent resource (for nesting purposes). If it is a top level, check out:
+ *                        [[TopLevelCourierCollectionResource]]
+ * @tparam K The key type of the resource.
+ * @tparam M The "value" type of the resource.
+ */
+abstract class CourierCollectionResource[ParentResource <: Resource[_], K, M <: ScalaRecordTemplate]()(
+    implicit kf: KeyFormat[K], classTag: ClassTag[M])
+  extends CollectionResource[ParentResource, K, M] {
+
+  final override implicit val keyFormat = kf
+
+  // When we use the serializer constructor, the classTag parameter is never initialized, and
+  // thus, we get NPEs when working with schemas. Because the OFormat isn't actually needed
+  final override implicit lazy val resourceFormat: OFormat[M] = Option(classTag).map { ct =>
+    CourierFormats.recordTemplateFormats[M](ct)
+  }.orNull // TODO: consider making a fake formatter.
+
+  protected[this] final lazy val BaseFields: Fields[M] = org.coursera.naptime.Fields(resourceFormat)
+
+  implicit override def Fields: Fields[M] = BaseFields
+}
+
+/**
+ * If the resource is a courier model, extending from [[TopLevelCourierCollectionResource]] removes
+ * some boilerplate.
+ *
+ * If you have a nested resource, check out [[CourierCollectionResource]].
+ *
+ * @param kf The [[KeyFormat]] for the key type in the resource.
+ * @param classTag The concrete class tag for the value type in the resource.
+ * @tparam K The key type of the resource.
+ * @tparam M The "value" type of the resource.
+ */
+abstract class TopLevelCourierCollectionResource[K, M <: ScalaRecordTemplate](
```

Comment:
can we make `TopLevelCourierCollectionResource` the default `CourierCollectionResource`, and either have a `NestedCourierCollectionResource` or some other way to override the `parentResource`? It seems like the top-level one is going to be used 99% of the time.